code/modules/factory/parts.dm

```diff
@@ -27,8 +27,9 @@
 /obj/item/factory_part/proc/advance_stage()
 	stage++
 	if(length(recipe) < stage)
-		GLOB.round_statistics.req_items_produced[result]++
-		new result(loc)
+		for(var/production AS in result)
+			GLOB.round_statistics.req_items_produced[production]++
+			new production(loc)
```

Comment:
this should support result as both list and typepath instead as initializing uneeded lists would be cpu waste

Reply:
Done